src/main/java/io/javalin/community/ssl/util/SSLUtils.java

```diff
@@ -50,18 +52,27 @@ public static SSLFactory getSslFactory(SSLConfig config, boolean reloading) {
         //Add the identity information
         parseIdentity(config, builder);
 
-        if (!reloading) builder.withSwappableIdentityMaterial();
+        //Add the trust information
+        if(config.getTrustConfig() != null) {
+            parseTrust(config.getTrustConfig(), builder);
+            builder.withNeedClientAuthentication();
+        }
 
-        builder.withSecurityProvider(config.securityProvider);
+        if (!reloading) {
+            builder.withSwappableIdentityMaterial();
```

Comment:
If reloading is set to false than on line 61 it will be negated and it will enter the if statement and enable reloading right?

Reply:
Yes that is correct.

Reply:
But that seems like a bug right? If the end-user disables it the code here enables it and if the end-user wants to enable it, it will ignore the property and not adjust the configuration in the sslfactory.

Reply:
I noticed you already resolved this one, but I am still curiious what your answer is? It would be a shame if it is a bug